### PR TITLE
create-login

### LINKED ISSUE #13: add sms login

### PR DESCRIPTION
**Pull request by Namaskar-1F64F**
closes #13

### CODE REVIEW

**Comment by RusseII** [comment]
Some of these issues are unrelated to this PR but i'll put them all down anyway. 

1. Can we improve the phone input? It took me a while to actually be able to enter my phone number.
2. Clicking "Create Excursion" doesn't update the map until a page refresh. Also - The behavior we'd probably want would be opening the provisions page when clicking on the create excursion button
3. This might be fixed when we make the excusurions page only the map, but sometimes I want to scroll up on the page but the map is selected and i'm unable to unselect the map and scroll to the top of the page
4. Way for users to log in if they get to landing page and already have an account? 
5. Excursions can be on top of eachother making them unclickable -  https://split-captain-git-create-login.sveny.vercel.app/#/voyages/ZtAy9NldWcrN3RCm49dH

**Comment by RusseII** [comment]
Also - you need to request me as a reviewer if you want me to see these.

image

File: src/Badge.svelte [hunk 1]
```diff
@@ -1,44 +0,0 @@
-<style>
```
Comment (RusseII) [inline src/Badge.svelte:1]: If you wanted to keep this hiding it on mobile only would be fine.

**Comment by Namaskar-1F64F** [comment]
I thought I had added you as a reviewer from the vscode interface, but I didn't press "OK" 

> Can we improve the phone input? It took me a while to actually be able to enter my phone number.

Yes, this was not intended behaviour for end users, just you.


> Clicking "Create Excursion" doesn't update the map until a page refresh. Also - The behavior we'd probably want would be opening the provisions page when clicking on the create excursion button

good idea, i couldn't figure out why the subscription wasn't updated, but going to the provisions page makes the most sense anyway.


> This might be fixed when we make the excusurions page only the map, but sometimes I want to scroll up on the page but the map is selected and i'm unable to unselect the map and scroll to the top of the page

yeah it's still not great, definitely needs fixed 


> Way for users to log in if they get to landing page and already have an account?

it should work. if you're logged in, it'll show you your profile.


> Excursions can be on top of eachother making them unclickable - split-captain-git-create-login.sveny.vercel.app/#/voyages/ZtAy9NldWcrN3RCm49dH

will be cleaned up on other excursions changes

File: src/itinerary/Itinerary.svelte [hunk 2]
```diff
@@ -3,12 +3,15 @@
...
 
-  export const addExcursion = () => {
-    $voyage.collection("excursions").doc().set({ name: "test" });
+  let excursionName = "Excursion Name";
+  const addExcursion = () => {
+    $voyage.collection("excursions").doc().set({ name: excursionName });
   };
 </script>
 
+<input bind:value={excursionName} />
```
Comment (RusseII) [inline src/itinerary/Itinerary.svelte]: You should have excursionName start out as an empty string and add a placeholder. 
RIght now the user has to delete the text "Excursion Name" before inputting the name

File: src/routes/Voyages.svelte [hunk 3]
```diff
@@ -1,32 +1,31 @@
...
   import { Collection, User } from "sveltefire";
   import { db } from "../firebase";
   import { link } from "svelte-spa-router";
 
-  export const addVoyage = () => {
-    db.collection("voyages").doc().set({ name: "YeeHaw Voyage" });
+  const addVoyage = () => {
+    db.collection("voyages").doc().set({ name: voyageName });
   };
+  let voyageName = "Voyage Name";
```
Comment (RusseII) [inline src/routes/Voyages.svelte]: You should have voyagename start out as an empty string and add a placeholder.
RIght now the user has to delete the text "Voyage name" before inputting the name

**Review by RusseII — CHANGES_REQUESTED** [review]

File: src/Badge.svelte (hunk 1 above)
Comment (Namaskar-1F64F) [reply src/Badge.svelte:1]: it doesn't make sense anymore if you're working on it

**Comment by RusseII** [comment]
Would you be able to make the commit messages more descriptive? It's hard to review each of the new commits when I don't know what the change is trying to fix. 

Thanks!

**Review by RusseII — APPROVED** [review]
Looks good to me. Can't test though as the vercel test domain isn't authorized to send text messages. 

Nice work with this. The input field for phone number is nice now.

**Comment by Namaskar-1F64F** [comment]
> Looks good to me. Can't test though as the vercel test domain isn't authorized to send text messages.
> 
> Nice work with this. The input field for phone number is nice now.

Should work on the https://split-captain-git-create-login.sveny.vercel.app/#/login domain.

**Comment by Namaskar-1F64F** [comment]
> Would you be able to make the commit messages more descriptive? It's hard to review each of the new commits when I don't know what the change is trying to fix.
> 
> Thanks!

I'm not sure that would follow the style-guide I have in my head, but if it's a detriment to the development of the project I could think about using the commit body as a compromise.

**Pull request merged**